scripts/tests/CRM/test_1b_decrypt_fidelity.py

```
@@ -209,12 +208,44 @@ def test_decrypt_and_verify(db_path: str, user_id: int, admin_privkey: str, orig
+
+    # Reject multi-statement input (internal semicolons)
+    if ";" in sql_normalized:
+        raise ValueError(f"run_docker_sql only allows single statements, got: {sql[:50]}")
+
+    # Validate: only allow SELECT statements (defense-in-depth for test helper)
+    if not sql_normalized.upper().startswith("SELECT"):
+        raise ValueError(f"run_docker_sql only allows SELECT statements, got: {sql[:50]}")
+
+    # Use list argv with stdin for SQL (no shell=True, no escaping needed)
+    result = subprocess.run(
+        ["docker", "compose", "exec", "-T", "backend", "sqlite3", db_path],
```

Ruff (0.14.14)

[error] Starting a process with a partial executable path

(S607)